File: commonfamily/src/main/java/com/koenig/communication/messages/TextMessage.java
package com.koenig.communication.messages;


import com.koenig.commonModel.Byteable;
import com.koenig.commonModel.Component;

import java.nio.ByteBuffer;

/**
 * Created by Thomas on 11.01.2017.
 */

public class TextMessage extends FamilyMessage {

    public static final String NAME = "TextMessage";
    public static final String UPDATE_FAIL = "update_fail";


    String text;

    public TextMessage(Component component, String text) {
        super(component);
        this.text = text;
    }

    public TextMessage(int version, Component component, String fromId, String toId, ByteBuffer buffer) {
        super(component);
        this.setVersion(version);
        this.setFromId(fromId);
        this.setToId(toId);
        this.text = Byteable.Companion.byteToString(buffer);
    }

    public String getText() {
        return text;
    }

    public String getName() {
        return NAME;
    }

    @Override
    protected int getContentLength() {
        return Byteable.Companion.getStringLength(text);
    }

    @Override
    protected void writeContent(ByteBuffer buffer) {
        buffer.put(Byteable.Companion.stringToByte(text));
    }

    @Override
    public String toString() {
        return "TextMessage{" +
                "text='" + text + '\'' +
                '}';
    }
}

File: commonfamily/src/main/java/com/koenig/communication/messages/TextMessage.kt
package com.koenig.communication.messages


import com.koenig.commonModel.Byteable
import com.koenig.commonModel.Component

import java.nio.ByteBuffer

/**
 * Created by Thomas on 11.01.2017.
 */

class TextMessage : FamilyMessage {


    var text: String
        internal set

    override val name: String
        get() = NAME

    override val contentLength: Int
        get() = Byteable.getStringLength(text)

    constructor(component: Component, text: String) : super(component) {
        this.text = text
    }

    constructor(version: Int, component: Component, fromId: String, toId: String, messageId: String, buffer: ByteBuffer) : super(component, messageId) {
        this.version = version
        this.fromId = fromId
        this.toId = toId
        this.text = Byteable.byteToString(buffer)
    }

    override fun writeContent(buffer: ByteBuffer) {
        buffer.put(Byteable.stringToByte(text))
    }

    override fun toString(): String {
        return "TextMessage{" +
                "text='" + text + '\''.toString() +
                '}'.toString()
    }

    companion object {

        val NAME = "TextMessage"
        val UPDATE_FAIL = "update_fail"
    }
}
